sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -585,6 +585,17 @@ object SQLConf {
       .longConf
       .createWithDefault(250 * 1024 * 1024)
 
+  val HIVE_FILESOURCE_FILE_CACHE_CONCURRENCY_LEVEL =
+    buildConf("spark.sql.hive.filesourceFileCacheConcurrencyLevel")
+      .doc("Provide concurrency level for underlying guava cache. The default is 1." +
+        " Higher value may give better performance, when the number of " +
```

Comment:
As it is set with default 1, `Higher value may give better performance` sounds like we set it to lowest performance.

Reply:
Is there any downside to allowing concurrency by default, even some large value? do we need another setting for this?

Reply:
We would like to see the evidence (i.e. multiple experimental results) to change the default value.   
Why do we need to change the default value in addition to making this parameter changeable?

Reply:
A large concurrency value has the following effect.
Size of each entry is limited by the value `totalCacheCapacity/concurrencyLevel` in guava cache. So if the value of `concurrencyLevel` is for example 4, then the maximum size of a single entry in the cache cannot be greater than 250MB/4 i.e. 62.5MB. 
Unless a user creates a HIVE table and setup partitions by specifying in the create table format, the entire cache is just a single entry. So, when listing content in a large directory, the requirement of cache is also large. This effect becomes evident, when listing a large directory of greater 100K items. See my other comment for how to reproduce.